src/rest_api/classes/person/widgets/lineage.clj

```diff
@@ -0,0 +1,349 @@
+    {:data (if (empty? data) nil data)
+     :description "people with whom this person worked"}))
+
+(def scalingHash
+  (json/parse-string (slurp "http://tazendra.caltech.edu/~azurebrd/cgi-bin/forms/wbpersonLineageScaling.json") true))
+
+(def roleColour
+  (hash-map
+    :Phd                            "#B40431"
+    :Postdoc                        "#00E300"
+    :Masters                        "#FF8000"
+    (keyword "Research staff")      "#08298A"
+    :Highschool                     "#05C1F0"
+    :Undergrad                      "#B58904"))
+
+(defn- parse-int [s]
+  (Integer. (re-find #"\d+" s)))
+
+(defn- generate-map-nodes-edges [oid]
+  (let [personid (get oid 0)
+        personname (str/replace (get oid 1) #"'" "")
+        otherpersonid (get oid 2)
+        otherpersonname (str/replace (get oid 3) #"'" "")
+        jsondata (get oid 4)
+        roles (filter #(re-find #"role" (str %)) (keys jsondata))
+        levels
+        (for [role roles
+              :let
+              [levels (str/capitalize
+                        (str/replace
+                          (str/replace
+                            (str/replace
+                              (str role) #":role/" "") #"-" " ") #"'" ""))]] levels)]
+    (for [level levels
+          data (if ((keyword level) roleColour) 
+                 (vector
+                   (hash-map
+                     :personid personid
+                     :personname personname
+                     :otherpersonid otherpersonid
+                     :otherpersonname otherpersonname
+                     :level level))
+                 nil )] data)))
+
+(defn- generate-map-this-person [oid]
+  (let [personid (get oid 0)
+        personname (get oid 1)
+        data (vector
+               (hash-map
+                 :personid personid
+                 :personname personname))] data))
+
+(defn- get-this-node-scaling [queriedData]
+  (let [value (first queriedData)
+        data  (if (nil? value)
+                1
+                (if (nil? ((keyword (:personid value)) scalingHash))
+                  1 
+                  (parse-int ((keyword (:personid value)) scalingHash))))] data))
+
+(defn- get-other-node-scaling [queriedData] 
+  (if (empty? queriedData) nil
+    (for [value queriedData
+          :let [data 
+                (if (nil? value)
+                  1
+                  (if (nil? ((keyword (:otherpersonid value)) scalingHash))
+                    1 
+                    (parse-int ((keyword (:otherpersonid value)) scalingHash))))]] data)))
+
+(defn- filter-this-node [directOrFull queriedData]
+  (let [value (first queriedData)
+        scaling (if (nil? ((keyword (:personid value)) scalingHash))
+                  1 
+                  (parse-int ((keyword (:personid value)) scalingHash)))
+        data (str "id: '" directOrFull (:personid value) "', name: '" (:personname value) "', url: '" (:personid value) "', scaling: '" scaling "', radius: '100', nodeshape: 'rectangle'")]
```

Comment:
split long line, e.g:
```clojure
  data (str "id : '"
                direct-or-full
                (:person-id value)
                ...)
```
Perhaps consider using `clojure.core/format` instead of `str` here?

Reply:
Fixed.  I looked into clojure.com/format, but it didn't seem to make it more readable.